firmware/Core/Src/antenna_deploy_drivers/ant_internal_drivers.c

```diff
@@ -0,0 +1,49 @@
+#include "main.h"
+
+#include "stm32l4xx_hal_i2c.h"
+#include "antenna_deploy_drivers/ant_internal_drivers.h"
+#include "debug_tools/debug_i2c.h"
+
+#include <stdint.h>
+#include <string.h>
+
+uint16_t ANT_ADDR = 0x31 << 1;
+
+extern I2C_HandleTypeDef hi2c2;
+
+// Might not need to include cmd buf array and length depending on the parameters of the command
+// If no parameters, you should only need to send the command byte (8 bits)
+
+/**
+ * @brief Sends a command to the antenna controller and waits for a response.
+ * 
+ * @param cmd_buf Array of bytes to send to the antenna controller
+ * @param cmd_len Length of the command buffer
+ * @param rx_buf Array to store the response from the antenna controller
+ * @param rx_len Length of the response buffer
+ * @return 0 upon success, 1 if tx_status received HAL_ERROR, 2 if tx_status received HAL_BUSY, 3 if tx_status received HAL_TIMEOUT, 4 if rx_status received HAL_ERROR
+ */
+uint8_t ANT_send_cmd_get_response(
```

Comment:
With so many commands having no response, I think it makes sense to split this function out into `ANT_send_cmd` and `ANT_get_response`. Allocating the `rx_buf` of length 0 is a little weird.

Reply:
Makes sense. I split the two functions up and changed the calling of the functions in ant_commands.c.